tower-util/src/lib.rs

```diff
@@ -76,6 +78,30 @@ pub trait ServiceExt<Request>: tower_service::Service<Request> {
     {
         CallAll::new(self, reqs)
     }
+
+    fn map_ok<F, Response>(self, f: F) -> MapOk<Self, F>
+    where
+        Self: Sized,
+        F: FnOnce(Self::Response) -> Response + Clone,
+    {
+        MapOk::new(self, f)
+    }
+
+    fn map_err<F, Error>(self, f: F) -> MapErr<Self, F>
+    where
+        Self: Sized,
+        F: FnOnce(Self::Error) -> Error + Clone,
+    {
+        MapErr::new(self, f)
+    }
+
+    fn with<F, NewRequest>(self, f: F) -> With<Self, F>
```

Comment:
The name `with` here is pulled from mirrored from here https://docs.rs/futures/0.3.1/futures/sink/trait.SinkExt.html#method.with

> Composes a function in front of the sink.
> 
> This adapter produces a new sink that passes each value through the given function f before sending it to self.

Reply:
I'm a little on the fence about naming...I agree that consistency with the `futures` crate is nice, but I don't find `with` to be super descriptive. To me, it sounds more like `layer`-like composition of two `Service`s. I _might_ consider erring on the side of consistency with the rest of _our_ API and calling this `MapRequest` or `MapReq`.

@LucioFranco what do you think?

Reply:
Not strictly a defense of `with`, but if we were to rename this `map_req` then https://github.com/tower-rs/tower/pull/396#discussion_r446291692 would imply the existence of a `try_map_req` (or something like that?), which is could be more of a mouthful than `try_with`.

Reply:
mmk, I'm fine with `with` if everyone else is!